File: app-java/src/main/java/csc435/app/FileRetrievalServer.java
package csc435.app;

public class FileRetrievalServer {
    public static void main( String[] args ) throws Exception
    {
        int port = Integer.parseInt(args[0]);
        IndexStore store = new IndexStore();
        ServerSideEngine engine = new ServerSideEngine(store);
        ServerAppInterface appInterface = new ServerAppInterface(engine);
        
        engine.initialize(port);
        appInterface.readCommands();
    }
}

File: app-cpp/src/file-retrieval-server.cpp
#include <iostream>
#include <memory>

#include "IndexStore.hpp"
#include "ServerSideEngine.hpp"
#include "ServerAppInterface.hpp"

int main(int argc, char** argv)
{
    std::shared_ptr<IndexStore> store = std::make_shared<IndexStore>();
    std::shared_ptr<ServerSideEngine> engine = std::make_shared<ServerSideEngine>(store);
    std::shared_ptr<ServerAppInterface> interface = std::make_shared<ServerAppInterface>(engine);

    // create a thread that creates and server TCP/IP socket and listenes to connections
    engine->initialize();

    // read commands from the user
    interface->readCommands();

    return 0;
}